llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -9529,7 +9647,21 @@ void VPWidenMemoryInstructionRecipe::execute(VPTransformState &State) {
   State.setDebugLocFrom(LI->getDebugLoc());
   for (unsigned Part = 0; Part < State.UF; ++Part) {
     Value *NewLI;
-    if (CreateGatherScatter) {
+    if (State.EVL) {
+      assert(State.UF == 1 &&
+             "Expected only UF==1 for predicated vectorization.");
+      Value *EVLPart = State.get(State.EVL, Part);
```

Comment:
Drop the part from EVLPart here as well? (going back to https://reviews.llvm.org/D99750?id=558054#inline-1551817))